library/Faktory/Ent/Batch.hs

```diff
@@ -0,0 +1,129 @@
+{-# LANGUAGE DerivingVia #-}
+
+-- | Support for the @BATCH@ command (Enterprise only)
+--
+-- <https://github.com/contribsys/faktory/wiki/Ent-Batches>
+--
+-- Batches allow multiple Jobs to be enqueued as a group, with a description
+-- (visible in the admin UI) and Jobs attached to run on completion of all Jobs
+-- within the group (always, or only if all were successful).
+--
+-- Usage:
+--
+-- @
+-- -- Build a Job to run at completion of the Batch. Arguments are the same as
+-- -- you would pass to 'perform' the Job.
+-- onComplete <- buildJob mempty producer myJob
+--
+-- 'runBatchT' ('complete' onComplete <> 'description' "My Batch") producer $ do
+--   -- Use 'batchPerform' instead of 'perform'
```

Comment:
We could make a type-class for `perform`. Then we could use it in both contexts:

```hs
-- Non-batch
perform :: HasPerform m => JobOptions -> Producer -> arg -> m JobId

-- Batched
perform :: HasPerform m => JobOptions -> Producer -> arg -> BatchT m JobId
```

But that's a pretty big API change, so I decided against for now. Happy to re-explore though.